scripts/illumina_run_parameter_parser.py

```diff
@@ -91,6 +93,75 @@ def attach_xml(process, run_dir):
                     sys.exit(2)
 
 
+def parse_illumina_interop(run_dir):
+    # Read interop
+    run_metrics = py_interop_run_metrics.run_metrics()
+    valid_to_load = py_interop_run.uchar_vector(py_interop_run.MetricCount, 0)
+    py_interop_run_metrics.list_summary_metrics_to_load(valid_to_load)
+    try:
+        run_metrics.read(run_dir, valid_to_load)
+    except Exception:
+        sys.stderr.write("Cannot parse information in InterOp")
+        sys.exit(2)
+    summary = py_interop_summary.run_summary()
+    py_interop_summary.summarize_run_metrics(run_metrics, summary)
+    lanes = summary.lane_count()
+    reads = summary.size()
+    # Parse the interop stats lane by lane for non-index reads
+    run_stats_summary = dict()
+    for lane in range(lanes):
+        lane_nbr = getattr(summary.at(0).at(lane), "lane")()
+        for read in range(reads):
+            if not summary.at(read).read().is_index():
+                stats = dict()
+                stats.update({"density"               :  getattr(summary.at(read).at(lane), "density")().mean()/1000})
+                stats.update({"error_rate"            :  getattr(summary.at(read).at(lane), "error_rate")().mean()})
+                stats.update({"first_cycle_intensity" :  getattr(summary.at(read).at(lane), "first_cycle_intensity")().mean()})
+                stats.update({"percent_aligned"       :  getattr(summary.at(read).at(lane), "percent_aligned")().mean()})
+                stats.update({"percent_gt_q30"        :  getattr(summary.at(read).at(lane), "percent_gt_q30")()})
+                stats.update({"percent_pf"           :  getattr(summary.at(read).at(lane), "percent_pf")().mean()})
+                stats.update({"phasing"               :  getattr(summary.at(read).at(lane), "phasing")().mean()})
+                stats.update({"prephasing"            :  getattr(summary.at(read).at(lane), "prephasing")().mean()})
+                stats.update({"reads_pf"              :  getattr(summary.at(read).at(lane), "reads_pf")()})
+                stats.update({"yield_g"               :  getattr(summary.at(read).at(lane), "yield_g")()})
+                if lane_nbr not in run_stats_summary.keys():
+                    run_stats_summary.update({lane_nbr: {read: stats}})
+                else:
+                    run_stats_summary[lane_nbr].update({read : stats})
+    return run_stats_summary
+
+
+def set_run_stats_in_lims(process, run_stats_summary):
+    for art in process.all_outputs():
+        if 'Lane' in art.name:
+            lane_nbr = int(art.name.split(' ')[1])
+            read = 1
+            for i in run_stats_summary[lane_nbr].keys():
+                if not math.isnan(run_stats_summary[lane_nbr][i]['density']):
```

Comment:
`run_stats_summary[lane_nbr][i]` assigned to a variable could improve readability as its used a lot of times.

Reply:
OK! I will refactor it!